frontends/PyCDE/src/pycde/module.py

```diff
@@ -394,6 +394,31 @@ def externmodule(to_be_wrapped, extern_name=None):
                                create_cb=create_msft_module_extern_op)
 
 
+def import_hw_module(hw_module: hw.HWModuleOp):
+  # Get the module name to use in the generated class and as the external name.
+  name = mlir.ir.StringAttr(hw_module.name).value
+
+  # Collect input and output ports as named Inputs and Outputs.
+  ports = {}
+  for input_name, block_arg in hw_module.inputs().items():
+    ports[input_name] = Input(block_arg.type, input_name)
+  for output_name, output_type in hw_module.outputs().items():
+    ports[output_name] = Output(output_type, output_name)
+
+  # Use the name and ports to construct a class object like what externmodule
+  # would wrap.
+  cls = type(name, (object,), ports)
+
+  # Creation callback that just moves the already build module into the System's
+  # ModuleOp and returns it.
+  def create_cb(sys: System, mod: _SpecializedModule, symbol: str):
```

Comment:
I _think_ you need to store `hw_module` as a default parameter to this function so it doesn't change with the next import_hw_module call. Maybe that's only for lambdas. Python closures suck.

Reply:
though based on your test below, perhaps not... I don't understand Python closures.

Reply:
yeah i think that weirdness is only for lambas.